paasta_tools/utils.py

```diff
@@ -3076,20 +3078,28 @@ def read_service_instance_names(
     service: str, instance_type: str, cluster: str, soa_dir: str
 ) -> Collection[Tuple[str, str]]:
     instance_list = []
-    conf_file = f"{instance_type}-{cluster}"
+    if instance_type == "spark":
+        conf_file = f"tron-{cluster}"
+    else:
+        conf_file = f"{instance_type}-{cluster}"
     config = service_configuration_lib.read_extra_service_information(
         service,
         conf_file,
         soa_dir=soa_dir,
         deepcopy=False,
     )
     config = filter_templates_from_config(config)
-    if instance_type == "tron":
+    if instance_type == "tron" or instance_type == "spark":
         for job_name, job in config.items():
-            action_names = list(job.get("actions", {}).keys())
-            for name in action_names:
-                instance = f"{job_name}.{name}"
-                instance_list.append((service, instance))
+            for action, action_info in job.get("actions", {}).items():
+                instance = f"{job_name}.{action}"
+                if (
+                    action_info.get("executor", "") == "ssh"
+                    and instance_type == "spark"
+                ):
```

Comment:
maybe this should be a function called `_is_spark_action`? and perhaps we want to also check if the command is spark-related since there are some ancient tronjobs that are practically unowned and still using ssh executors for non-spark stuff